homeassistant/components/intellifire/sensor.py

```diff
@@ -85,6 +86,42 @@ def _time_remaining_to_timestamp(data: IntellifirePollData) -> datetime | None:
         device_class=SensorDeviceClass.TIMESTAMP,
         value_fn=_time_remaining_to_timestamp,
     ),
+    IntellifireSensorEntityDescription(
+        key="downtime",
+        name="Downtime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.downtime,
+    ),
+    IntellifireSensorEntityDescription(
+        key="uptime",
+        name="Uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.uptime,
+    ),
+    IntellifireSensorEntityDescription(
+        key="connection_quality",
+        name="Connection Quality",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.connection_quality,
+    ),
+    IntellifireSensorEntityDescription(
+        key="ecm_latency",
+        name="ECM Latency",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.ecm_latency,
+    ),
+    IntellifireSensorEntityDescription(
+        key="ipv4_address",
+        name="IP",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.ipv4_address,
+    ),
+    IntellifireSensorEntityDescription(
+        key="serial",
+        name="Serial",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.serial,
```

Comment:
Remove this one. It is way too long and doesn't add value? Users can see it in their app if they need it.